examples/demo-rollup/src/txs_rpc.rs

```diff
@@ -0,0 +1,63 @@
+use jsonrpsee::RpcModule;
+// use serde::de::DeserializeOwned;
+// use serde::Serialize;
+use sov_rollup_interface::services::batch_builder::BatchBuilder;
+use sov_rollup_interface::services::da::DaService;
+use std::sync::{Arc, RwLock};
+
+/// Single data structure that manages mempool and batch producing.
+pub struct TxsRpcHandler<B: BatchBuilder, T: DaService> {
+    batch_builder: RwLock<B>,
+    da_service: Arc<T>,
+}
+
+impl<B: BatchBuilder + Send + Sync, T: DaService + Send + Sync> TxsRpcHandler<B, T> {
+    pub fn new(batch_builder: B, da_service: Arc<T>) -> Self {
+        Self {
+            batch_builder: RwLock::new(batch_builder),
+            da_service,
+        }
+    }
+
+    async fn submit_batch(&self) -> Result<(), anyhow::Error> {
+        // Need to release lock before await, so Future is `Send`.
+        let blob = {
+            let mut batch_builder = self.batch_builder.write().unwrap();
+            batch_builder.get_next_blob()?
+        };
+        let blob: Vec<u8> = blob.into_iter().flatten().collect();
+        self.da_service.send_transaction(&blob).await.unwrap();
+        Ok(())
+    }
+}
+
+fn register_txs_rpc_methods<B, D>(
+    rpc: &mut RpcModule<TxsRpcHandler<B, D>>,
+) -> Result<(), jsonrpsee::core::Error>
+where
+    // R: Serialize + DeserializeOwned,
+    // T: Serialize + DeserializeOwned,
+    B: BatchBuilder + Send + Sync + 'static,
+    D: DaService + Send + Sync + 'static,
+{
+    rpc.register_async_method("submit_batch", |_, batch_builder| async move {
```

Comment:
The JsonRPC convention is `{namespace}_{methodName}`. Maybe we want the namespace to be `builder`? So something like `builder_submitBatch`?

Reply:
I agree, it needs follow that pattern.

Isn't `builder` too generic? What if it is `batchBuilder_submit`? 

I use term `mempool` everywhere, but I am not sure if this is correct in this case, but it can be `mempool_produceBatch` But then whole `BatchBuilder` trait probably needs to be renamed. Which is not a big problem at the moment.

Maybe side question, how do you envision mempool in the full node, would it be some completely separate module?

Reply:
Decided on a call